internal/services/controllers/v1/olap/controller.go

```diff
@@ -646,7 +672,76 @@ func (tc *OLAPControllerImpl) handleCreateMonitoringEvent(ctx context.Context, t
 		opts = append(opts, event)
 	}
 
-	return tc.repo.OLAP().CreateTaskEvents(ctx, tenantId, opts)
+	err = tc.repo.OLAP().CreateTaskEvents(ctx, tenantId, opts)
+
+	if err != nil {
+		return err
+	}
+
+	if !tc.repo.OLAP().PayloadStore().ExternalStoreEnabled() {
+		return nil
+	}
+
+	offloadToExternalOpts := make([]v1.OffloadToExternalStoreOpts, 0)
+	idInsertedAtToExternalId := make(map[v1.IdInsertedAt]pgtype.UUID)
+
+	for _, opt := range opts {
+		// generating a dummy id + inserted at to use for creating the external keys for the task events
+		// we do this since we don't have the id + inserted at of the events themselves on the opts, and we don't
+		// actually need those for anything once the keys are created.
+		dummyId := rand.Int63()
+		// randomly jitter the inserted at time by +/- 300ms to make collisions virtually impossible
+		dummyInsertedAt := time.Now().Add(time.Duration(rand.Intn(2*300+1)-300) * time.Millisecond)
+
+		idInsertedAtToExternalId[v1.IdInsertedAt{
+			ID:         dummyId,
+			InsertedAt: sqlchelpers.TimestamptzFromTime(dummyInsertedAt),
+		}] = opt.ExternalID
+
+		offloadToExternalOpts = append(offloadToExternalOpts, v1.OffloadToExternalStoreOpts{
+			StorePayloadOpts: &v1.StorePayloadOpts{
+				Id:         dummyId,
+				InsertedAt: sqlchelpers.TimestamptzFromTime(dummyInsertedAt),
+				ExternalId: opt.ExternalID,
+				Type:       sqlcv1.V1PayloadTypeTASKEVENTDATA,
+				Payload:    opt.Output,
+				TenantId:   tenantId,
```

Comment:
this logic is really janky - we have to do the same thing for the events too. the basic issue is they will never be updated from the core db, so we need to do the offloading / cutovers on the OLAP side. the issue is that if we want to respect the same TTL as in the core repo, we'd need another set of WAL tables here which IMO is overkill.

@abelanger5 WDYT of this? I'm not sure if there's another alternative other than either something like this or having WAL tables for these in the OLAP code